### PR TITLE
Problem: Missing generic configuration

### LINKED ISSUE #657: Problem: Missing generic configuration
Currently, configurations are highly specific e.g. toml tags and variable from env. We could make this to be more generic for variety of implementation ways.

### PR DESCRIPTION
**Pull request by davcrypto**
Resolved #657 , #623 and #648

### CODE REVIEW

File: example/app/example-app/projections.go [hunk 1]
```diff
@@ -313,7 +315,13 @@ func InitProjection(name string, params InitProjectionParams) projection_entity.
...
 		migrationHelper := github_migrationhelper.NewGithubMigrationHelper(sourceURL, databaseURL)
 
-		return bridge_pending_activity.NewBridgePendingActivity(params.Logger, params.RdbConn, migrationHelper)
+		config := bridge_pending_activity.Config{}
+		err := config.Fill(params.ExtraConfigs[name])
+		if err != nil {
+			params.Logger.Panicf(err.Error())
+		}
+
+		return bridge_pending_activity.NewBridgePendingActivity(params.Logger, params.RdbConn, migrationHelper, bridge_pending_activity.Config{})
```
Comment (calvinaco) [inline example/app/example-app/projections.go:324]: Are we supposed to use `config` here?

File: example/app/example-app/run.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func run(args []string) error {
 
 			&cli.StringFlag{
 				Name:    "tendermintURL",
-				Usage:   "Tendermint HTTP RPC URL",
+				Usage:   "TendermintApp HTTP RPC URL",
```
Comment (calvinaco) [inline example/app/example-app/run.go:81]: Is this change on purpose?

File: example/app/example-app/cronjobs.go [hunk 3]
```diff
@@ -58,7 +60,13 @@ func InitCronJob(name string, params InitCronJobParams) projection_entity.CronJo
 		databaseURL := migrationhelper.GenerateDefaultDatabaseURL(name, connString)
 		migrationHelper := github_migrationhelper.NewGithubMigrationHelper(sourceURL, databaseURL)
 
-		return bridge_activity_matcher.New(params.Logger, params.RdbConn, migrationHelper)
+		config := bridge_activity_matcher.Config{}
+		err := config.Fill(params.ExtraConfigs[name])
```
Comment (calvinaco) [inline example/app/example-app/cronjobs.go:64]: If the use of `Fill` is expected to be followed by first allocating an empt struct, how about make it a new method?

`bridge_activity_matcher.DecodeConfigFromInterface(params.ExtraConfigs[name])`

File: projection/bridge_activity/bridge_activity_matcher/bridge_activity_matcher.go [hunk 4]
```diff
@@ -37,6 +38,27 @@ type Config struct {
...
+		return fmt.Errorf("error creating cronjob config decoder: %v", decoderErr)
+	}
+
+	if err := decoder.Decode(data); err != nil {
+		return fmt.Errorf("error decoding cronjob BridgeActivityMatcher config: %v", err)
+	}
+
+	return nil
+}
+
```
Comment (calvinaco) [inline projection/bridge_activity/bridge_activity_matcher/bridge_activity_matcher.go:61]: Love the way for each projection to be self-contained on config reading.

File: example/app/example-app/projections.go (hunk 1 above)
Comment (davcrypto) [reply example/app/example-app/projections.go]: Good Catch! Updated

File: example/app/example-app/run.go (hunk 2 above)
Comment (davcrypto) [reply example/app/example-app/run.go]: Should be the IDE refactor effect. Changed back.

File: example/app/example-app/cronjobs.go (hunk 3 above)
Comment (davcrypto) [reply example/app/example-app/cronjobs.go]: Thanks for the advice, Updated

**Review by ysong42 — APPROVED** [review]
This is so cool! LGTM!
